src/shiftml/ase/calculator.py

```diff
@@ -0,0 +1,143 @@
+import os
+import urllib.request
+
+from metatensor.torch.atomistic import ModelOutput
+from metatensor.torch.atomistic.ase_calculator import MetatensorCalculator
+
+url_resolve = {
+    "ShiftML1.0": "https://tinyurl.com/3xwec68f",
+    "ShiftML1.1": "https://tinyurl.com/53ymkhvd",
+}
+
+resolve_outputs = {
+    "ShiftML1.0": {"mtt::cs_iso": ModelOutput(quantity="", unit="ppm", per_atom=True)},
+    "ShiftML1.1": {"mtt::cs_iso": ModelOutput(quantity="", unit="ppm", per_atom=True)},
+}
+
+resolve_fitted_species = {
+    "ShiftML1.0": set([1, 6, 7, 8, 16]),
+    "ShiftML1.1": set([1, 6, 7, 8, 16]),
+}
+
+
+class ShiftML(MetatensorCalculator):
+    """
+    ShiftML calculator for ASE
+    """
+
+    def __init__(self, model_version, force_download=False):
+        """
+        Initialize the ShiftML calculator
+
+        Parameters
+        ----------
+        model_version : str
+            The version of the ShiftML model to use. Supported versions are
+            "ShiftML1.0".
+        force_download : bool, optional
+            If True, the model will be downloaded even if it is already in the cache.
+            Default is False.
+        """
+
+        try:
+            # The rascline import is necessary because
+            # it is required for the scripted model
+            import rascaline.torch
+
+            print(rascaline.torch.__version__)
+            print("rascaline-torch is installed, importing rascaline-torch")
+
+        except ImportError:
+            raise ImportError(
+                "rascaline-torch is required for ShiftML calculators,\
+                 please install it using\
+                 pip install git+https://github.com/luthaf/rascaline#subdirectory\
+                 =python/rascaline-torch"
+            )
+
+        try:
+            url = url_resolve[model_version]
+            self.outputs = resolve_outputs[model_version]
+            self.fitted_species = resolve_fitted_species[model_version]
+            print("Found model version in url_resolve")
+            print("Resolving model version to model files at url: ", url)
+        except KeyError:
+            raise ValueError(
+                f"Model version {model_version} is not supported.\
+                    Supported versions are {list(url_resolve.keys())}"
+            )
+
+        cachedir = os.path.expanduser(
+            os.path.join("~", ".cache", "shiftml", str(model_version))
+        )
```

Comment:
instead of ding this which will work fine only on linux, you could use https://pypi.org/project/platformdirs/ and its `user_cache_dir`. 

This should also respect user configuration such as XDG_CACHE_HOME (from https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html)